pkg/cmd/drt-run/operations.go

```diff
@@ -109,9 +109,11 @@ func (r *opsRunner) pickOperation(
 		opSpecsSet := r.specs[setIdx]
 		opSpec := &opSpecsSet[rng.Intn(len(opSpecsSet))]
 		r.mu.Lock()
+		defer func() {
+			r.mu.Unlock()
```

Comment:
This will run at the end of the function, not at the end of the loop, so we'll be doing double-locks and waiting forever. You'll need to rewrite this to have a function wrapping everything inside the mutex-excluded part of the code.